halibot/halauth.py

```diff
@@ -1,11 +1,21 @@
 import json
 import logging
+from halibot.message import Message
 
 def hasPermission(perm, reply=False):
 	def real_dec(func):
-		def wrapper(self, msg, *args, **kwargs):
+		def wrapper(self, *args, **kwargs):
+			msg = None
+			for i in list(args) + list(kwargs.values()):
+				if i.__class__ == Message:
```

Comment:
Not very pythonic, what about subclasses of `Message`?

Reply:
Also, what should we do in the cases with multiple message arguments? I don't think just taking the first one that appears is the most transparent behavior. Maybe an argument to the decorator to specify which to use?

Reply:
Agh, that should be a subclass of `Message` check not a equality check. I blame hacktoberfest-rush @richteer on that laziness.

As for multiple message arguments: The previous convention stated that the message argument should be first, and I agree that should be the case now as well. That said, that means this wrapper doesn't get any easier to use if you still have to uphold the convention, so convenience is lost I suppose.

Alternative suggestion: Have two different wrappers:
1. One that requires a `Message` object as the first parameter
2. One that requires exactly *one* `Message` object in `*args`

Reply:
Maybe for simplicity we just have one wrapper with the general use case, and when people need to do fancy things, like pass multiple messages around, they just have to call `hasPermission` themself?

Reply:
Probably the better choice. I'm leaning towards requiring the message to be in `**kwargs`, as `msg`, since that is what `CommandModule` uses as convention.

Reply:
I'm fine with that. More clear than first positional.